lib/active_merchant/billing/gateways/firstdata_e4_v27.rb

```diff
@@ -0,0 +1,445 @@
+        xml = Builder::XmlMarkup.new
+
+        add_amount(xml, money, options)
+
+        if credit_card_or_store_authorization.is_a? String
+          add_credit_card_token(xml, credit_card_or_store_authorization, options)
+        else
+          add_credit_card(xml, credit_card_or_store_authorization, options)
+        end
+
+        add_address(xml, options)
+        add_customer_data(xml, options)
+        add_invoice(xml, options)
+        add_tax_fields(xml, options)
+        add_level_3(xml, options)
+
+        xml.target!
+      end
+
+      def build_capture_or_credit_request(money, identification, options)
+        xml = Builder::XmlMarkup.new
+
+        add_identification(xml, identification)
+        add_amount(xml, money, options)
+        add_customer_data(xml, options)
+        add_card_authentication_data(xml, options)
+
+        xml.target!
+      end
+
+      def build_store_request(credit_card, options)
+        xml = Builder::XmlMarkup.new
+
+        add_credit_card(xml, credit_card, options)
+        add_address(xml, options)
+        add_customer_data(xml, options)
+
+        xml.target!
+      end
+
+      def add_credentials(xml)
+        xml.tag! 'ExactID', @options[:login]
+        xml.tag! 'Password', @options[:password]
+      end
+
+      def add_transaction_type(xml, action)
+        xml.tag! 'Transaction_Type', TRANSACTIONS[action]
+      end
+
+      def add_identification(xml, identification)
+        authorization_num, transaction_tag, _ = identification.split(';')
+
+        xml.tag! 'Authorization_Num', authorization_num
+        xml.tag! 'Transaction_Tag', transaction_tag
+      end
+
+      def add_amount(xml, money, options)
+        currency_code = options[:currency] || default_currency
+        xml.tag! 'DollarAmount', localized_amount(money, currency_code)
+        xml.tag! 'Currency', currency_code
+      end
+
+      def add_credit_card(xml, credit_card, options)
+        if credit_card.respond_to?(:track_data) && credit_card.track_data.present?
+          xml.tag! 'Track1', credit_card.track_data
+          xml.tag! 'Ecommerce_Flag', 'R'
+        else
+          xml.tag! 'Card_Number', credit_card.number
+          xml.tag! 'Expiry_Date', expdate(credit_card)
+          xml.tag! 'CardHoldersName', credit_card.name
+          xml.tag! 'CardType', card_type(credit_card.brand)
+
+          add_credit_card_eci(xml, credit_card, options)
+          add_credit_card_verification_strings(xml, credit_card, options)
+        end
+      end
+
+      def add_credit_card_eci(xml, credit_card, options)
+        eci = if credit_card.is_a?(NetworkTokenizationCreditCard) && credit_card.source == :apple_pay && card_brand(credit_card) == 'discover'
+          # Discover requires any Apple Pay transaction, regardless of in-app
+          # or web, and regardless of the ECI contained in the PKPaymentToken,
+          # to have an ECI value explicitly of 04.
+          '04'
+        else
```

Comment:
Rubocop unhappy with indentation of `else` relative to `if`. Open to ideas on how best to make it happy.

Reply:
It wants the `else` lined up with the `if` (and will in general want the whole block appropriately indented), so line 224 should start two spaced after the `if`, and the `else` should align with the `if` also.

Reply:
You can run `rubocop --auto-correct` to see what it suggests, BTW.